sqlglot/dialects/dialect.py

```diff
@@ -37,6 +37,79 @@
 }
 
 
+DATE_PART_MAPPING = {
```

Comment:
@VaggelisD is this the same mapping as in Snowflake? Did you double check if all units are included? https://docs.aws.amazon.com/redshift/latest/dg/r_Dateparts_for_datetime_functions.html

Reply:
I added the missing ones. A change in plans as I noticed that Snowflake mapped `"epoch" -> "epoch_seconds"`, but in Redshift it's `"epoch" -> "epoch"`. I'm guessing there might be differences across dialects, so generalizing among them became a bit harder..

Reply:
I tried that but the builder functions that `map_date_part` is applied are outside of dialect. The same happens with the `TimeUnit` refactor we had in mind, it doesn't have access to dialect at `expressions.py`

Reply:
The builder functions have access to the dialect, you just need to specify it - it's an optional arg. So, `exp.SomeExpr: lambda args, dialect: ...`. Does that help?

Reply:
Hmm, will check it out. I think it will make things a bit more troublesome than the current version though, with the builder `dialect` boilerplate

Reply:
If it turns out too complicated we can delay implementing a more robust / generic solution until a larger scale date part refactor is done.